src/app/redux/GlobalReducer.js

```diff
@@ -1,302 +1,424 @@
 import {Map, Set, List, fromJS, Iterable} from 'immutable';
-import createModule from 'redux-modules';
 import {emptyContent} from 'app/redux/EmptyState';
+import {contentStats} from 'app/utils/StateFunctions';
 import constants from './constants';
-import {contentStats} from 'app/utils/StateFunctions'
-
-const emptyContentMap = Map(emptyContent)
-
-export default createModule({
-    name: 'global',
-    initialState: Map({status: {}}),
-    transformations: [
-        {
-            action: 'SET_COLLAPSED',
-            reducer: (state, action) => {
-                return state.withMutations(map => {
-                    map.updateIn(['content', action.payload.post], value => {
-                        value.merge(Map({collapsed: action.payload.collapsed}));
-                    });
+
+const emptyContentMap = Map(emptyContent);
+
+const defaultState = Map({status: {}});
+
+// Action constants
+const RECEIVE_STATE = 'global/RECEIVE_STATE';
+const SET_COLLAPSED = 'global/SET_COLLAPSED';
+const RECEIVE_ACCOUNT = 'global/RECEIVE_ACCOUNT';
+const RECEIVE_COMMENT = 'global/RECEIVE_COMMENT';
+const RECEIVE_CONTENT = 'global/RECEIVE_CONTENT';
+const LINK_REPLY = 'global/LINK_REPLY';
+const UPDATE_ACCOUNT_WITNESS_VOTE = 'global/UPDATE_ACCOUNT_WITNESS_VOTE';
+const UPDATE_ACCOUNT_WITNESS_PROXY = 'global/UPDATE_ACCOUNT_WITNESS_PROXY';
+const DELETE_CONTENT = 'global/DELETE_CONTENT';
+const VOTED = 'global/VOTED';
+const FETCHING_DATA = 'global/FETCHING_DATA';
+const RECEIVE_DATA = 'global/RECEIVE_DATA';
+const RECEIVE_RECENT_POSTS = 'global/RECEIVE_RECENT_POSTS';
+const REQUEST_META = 'global/REQUEST_META';
+const RECEIVE_META = 'global/RECEIVE_META';
+const SET = 'global/SET';
+const REMOVE = 'global/REMOVE';
+const UPDATE = 'global/UPDATE';
+const SET_META_DATA = 'global/SET_META_DATA';
+const CLEAR_META = 'global/CLEAR_META';
+const CLEAR_META_ELEMENT = 'global/CLEAR_META_ELEMENT';
+const FETCH_JSON = 'global/FETCH_JSON';
+const FETCH_JSON_RESULT = 'global/FETCH_JSON_RESULT';
+const SHOW_DIALOG = 'global/SHOW_DIALOG';
+const HIDE_DIALOG = 'global/HIDE_DIALOG';
+// Saga-related:
+export const GET_STATE = 'global/GET_STATE';
+
+export default function reducer(state = defaultState, action) {
+    const payload = action.payload;
+
+    switch (action.type) {
+        case SET_COLLAPSED: {
```

Comment:
Using a switch block like this is koo (if/else). I'd suggest alphabetizing the `case:` params. It'll make it easier to use, and might actually make it easier to review. (unless there are perf-reasons or non- `return;' / 'break;` blocks.

Reply:
i kept the original order. i don't want to get too deep into the weeds of totally rewriting the reducers at this point; open an issue for this one

Reply:
Added #2072